### DISCUSSION TITLE
How to disable a certain gesture?

### DISCUSSION THREAD

**Discussion by mlshv** [cytoscape/cytoscape.js#3165] [Q&A]
The doc for [Gestures](https://js.cytoscape.org/#notation/gestures) says:
> All gesture actions can be controlled by the programmer, toggling them on or off whenever needed.

However I wasn't able to find a way to disable a certain gesture.

I want to disable "Two finger trackpad up or down to zoom" gesture to make it possible to pan using two finger trackpad swipe (like Figma does). I tried this, but no success:

```js
cy.container().addEventListener('wheel', function(e) {
  if (e.ctrlKey) {
    return; // Allow default zoom behavior when control key is pressed
  }

  // Prevent the default zoom behavior — this DOES NOT work
  e.preventDefault();

  const pan = cy.pan();
  const deltaY = e.deltaY;
  const deltaX = e.deltaX;

  // Update the Cytoscape pan
  cy.pan({
    x: pan.x - deltaX,
    y: pan.y - deltaY
  });
});
```

**Comment by maxkfranz** [comment]
This sounds more like a new feature than disabling an existing gesture.  How would this proposed feature work in Cytoscape, where panning can happen in every direction?

**Reply by mlshv** [reply]
The thing that i'm trying to achieve may indeed sound like a new feature. However, according to the existing documentation, it should be possible to implement it without altering the Cytoscape source code.

Please refer to the following demo for clarification: https://codepen.io/mlshv-the-bold/full/MWZvvLy

I've also attached a screen recording in case you don't have a trackpad.
https://github.com/cytoscape/cytoscape.js/assets/16861579/61bab689-8cc9-4622-9b89-f95e347ab184

The demo shows that panning with the trackpad functions as expected, yet executing a similar two-finger scroll gesture results in zooming.

Ideally, there would be an option to disable the "Two-finger trackpad up or down to zoom" gesture.

Therefore, the question is: is there a way to disable this specific gesture? Or does the documentation contain an inaccurate statement when it says "All gesture actions can be controlled by the programmer, toggling them on or off whenever needed"?

**Reply by maxkfranz** [reply]
> Ideally, there would be an option to disable the "Two-finger trackpad up or down to zoom" gesture.

(1) You can indeed disable user zooming gestures: https://js.cytoscape.org/#cy.userZoomingEnabled

(2) The browser reports two-finger trackpad events as wheel events.  If you want a new feature to allow for two-finger scrolling, you'd need to create a new panning gesture with wheel events.  That may be a better fit for an extension, since it is a bit niche and it does not support all devices (e.g. users with a scroll wheel mouse would be out of luck).
